xcp_d/workflow/anatomical.py

```diff
@@ -1,784 +1,1371 @@
-                                        n_procs=omp_nthreads)
-            brainspritex_wf = pe.Node(BrainPlotx(),
-                                      name='brainsprite',
-                                      mem_gb=mem_gb,
-                                      n_procs=omp_nthreads)
-            ds_brainspriteplot_wf = pe.Node(DerivativesDataSink(
-                base_directory=output_dir,
-                check_hdr=False,
-                dismiss_entities=[
-                    'desc',
-                ],
-                desc='brainplot',
-                datatype="figures"),
-                name='brainspriteplot')
-
-            workflow.connect([
-                (inputnode, enhancet1w_wf, [('t1w', 'in_file')]),
-                (enhancet1w_wf, brainspritex_wf, [('out_file', 'template')]),
-                (inputnode, ribbon2statmap_wf, [('t1seg', 'ribbon')]),
-                (ribbon2statmap_wf, brainspritex_wf, [('out_file', 'in_file')
-                                                      ]),
-                (brainspritex_wf, ds_brainspriteplot_wf, [('out_html',
-                                                           'in_file')]),
-                (inputnode, ds_brainspriteplot_wf, [('t1w', 'source_file')]),
-            ])
-
-    return workflow
-
-
-def _getsesid(filename):
-    ses_id = None
-    filex = os.path.basename(filename)
-
-    file_id = filex.split('_')
-    for k in file_id:
-        if 'ses' in k:
-            ses_id = k.split('-')[1]
-            break
-
-    return ses_id
+# emacs: -*- mode: python; py-indent-offset: 4; indent-tabs-mode: nil -*-
+# vi: set ft=python sts=4 ts=4 sw=4 et:
+"""
+fectch anatomical files/resmapleing surfaces to fsl32k
+^^^^^^^^^^^^^^^^^^^^^^^^
+.. autofunction:: init_structral_wf
+
+"""
+
+import os
+import fnmatch
+import shutil
+from pathlib import Path
+from templateflow.api import get as get_template
+from ..utils import collect_data, CiftiSurfaceResample
+from nipype.interfaces.freesurfer import MRIsConvert
+from ..interfaces.connectivity import ApplyTransformsx
+from niworkflows.engine.workflows import LiterateWorkflow as Workflow
+from pkg_resources import resource_filename as pkgrf
+from nipype.pipeline import engine as pe
+from nipype.interfaces import utility as niu
+from ..interfaces import BrainPlotx, RibbontoStatmap
+from ..utils import bid_derivative
+from nipype.interfaces.ants.resampling import (
+    ApplyTransforms as antsapplytransforms,
+)  # TM
+from nipype.interfaces.ants import CompositeTransformUtil  # MB
+from nipype.interfaces.fsl.maths import BinaryMaths as fslbinarymaths  # TM
+from nipype.interfaces.fsl import Merge as fslmerge  # TM
+from ..interfaces.c3 import C3d  # TM
+from ..interfaces.ants import CompositeInvTransformUtil  # TM
+from ..interfaces.workbench import (
+    ApplyWarpfield,
+    SurfaceGenerateInflated,
+    SurfaceAverage,
+)  # MB,TM
+
+
+class DerivativesDataSink(bid_derivative):

```

Comment:
This class keeps on getting redefined in various places. Note to self: we should remove these redefinitions